src/assignment/assignment.service.ts

```diff
@@ -1573,10 +1574,14 @@ export class AssignmentService {
 	): Promise<void> {
 		try {
 			await fetchWithLogoutJson(
-				`${getEnv('PROXY_URL')}/assignments/${assignmentId}/share/add-contributor?email=${
-					user.email
-				}&rights=${user.rights}`,
-				{ method: 'PATCH' }
+				stringifyUrl({
+					url: `${getEnv('PROXY_URL')}/assignments/${assignmentId}/share/add-contributor`,
+					query: {
+						email: user.email || '',
```

Comment:
might be better to throw an exception if user.mail is empty, so we don't add any contributors with an empty email